polkadot/node/core/pvf/execute-worker/src/lib.rs

```diff
@@ -145,17 +113,28 @@ pub fn worker_entrypoint(
 		worker_dir_path,
 		node_version,
 		worker_version,
-		|mut stream, worker_dir_path| {
-			let worker_pid = process::id();
-			let artifact_path = worker_dir::execute_artifact(&worker_dir_path);
+		|mut stream, worker_info| {
+			let artifact_path = worker_dir::execute_artifact(&worker_info.worker_dir_path);
 
 			let Handshake { executor_params } = recv_execute_handshake(&mut stream)?;
 
+			let executor_params: Arc<ExecutorParams> = Arc::new(executor_params);
+			let execute_worker_stack_size = get_max_stack_size(&executor_params, 1);
+			// Number of threads here is 3:
+			// 1 - Main thread
+			// 2 - Cpu monitor thread
+			// 3 - Execute thread
+			//
+			// NOTE: if the number of threads inside the child process change in the future, this
+			// value must be changed as well.
```

Comment:
Let's move `3` to a constant (see my other comment) and move this comment to a docstring over the new constant. And please add, "The correctness of this value is enforced by a test".